src/main/java/org/kohsuke/github/GitHubClient.java

```diff
@@ -112,25 +108,26 @@
         this.connector = connector;
 
         if (oauthAccessToken != null) {
-            encodedAuthorization = "token " + oauthAccessToken;
+            this.credentialProvider = new ImmutableCredentialProvider(String.format("token %s", oauthAccessToken));
```

Comment:
```suggestion
            this.credentialProvider = ImmutableCredentialProvider.fromOAuthToken(String.format("token %s", oauthAccessToken));
```

Reply:
I would also suggest moving all these into `GitHubBuilder` and have a method `GitHubBuilder.withCredentialProvider(CredentialProvider)` which can be internal for now.  Then this constructor just sets `this.credentialProvider = credentialProvider` pass from the builder.